platform/qt_location_service.cpp

```diff
@@ -0,0 +1,172 @@
+
+  if (!isConnected)
+  {
+    LOG(LERROR, ("Failed to connect the signal:", "errorOccurred"));
+    return;
+  }
+  LOG(LDEBUG, ("Signal successfully connected:", "errorOccurred"));
+
+  isConnected = (bool)connect(m_positionSource, &QGeoPositionInfoSource::supportedPositioningMethodsChanged,
+      this, &QtLocationService::OnSupportedPositioningMethodsChanged);
+
+  if (!isConnected)
+  {
+    LOG(LERROR, ("Failed to connect the signal:", "supportedPositioningMethodsChanged"));
+    return;
+  }
+  LOG(LDEBUG, ("Signal successfully connected:", "supportedPositioningMethodsChanged"));
+
+  m_positionSource->setUpdateInterval(1000);
+  m_positionSource->setPreferredPositioningMethods(QGeoPositionInfoSource::AllPositioningMethods);
+}
+
+void QtLocationService::OnLocationUpdate(const QGeoPositionInfo &info)
+{
+  if (!info.isValid())
+  {
+    LOG(LWARNING, ("Location update with Invalid timestamp or coordinates from:", m_positionSource->sourceName().toStdString()));
+  }
+  const auto& coordinate = info.coordinate();
+  LOG(LDEBUG, ("Location updated with valid coordinates:", coordinate.longitude(), coordinate.latitude()));
+  m_observer.OnLocationUpdated(gpsInfoFromQGeoPositionInfo(info, qStringToTLocationSource(m_positionSource->sourceName())));
+}
+
+void QtLocationService::OnErrorOccurred(QGeoPositionInfoSource::Error positioningError)
+{
+  LOG(LERROR, ("Location error occured QGeoPositionInfo::Error code:", positioningError));
+  m_observer.OnLocationError(tLocationErrorFromQGeoPositionInfoError(positioningError));
+}
+
+void QtLocationService::OnSupportedPositioningMethodsChanged()
+{
+  LOG(LWARNING, ("Supported Positioning Method changed:", m_positionSource->sourceName().toStdString()));
+  // In certain cases propagating GPSIsOff would make sense,
+  // but this signal can also mean GPSISOn ... or something entirely different.
+  // m_observer.OnLocationError(location::TLocationError::EGPSIsOff);
+}
+
+void QtLocationService::Start()
+{
+  if (!m_positionSource)
+  {
+    LOG(LWARNING, ("No supported source is available for starting the positioning with:", m_positionSource->sourceName().toStdString()));
+  }
+  else
+  {
+    LOG(LDEBUG, ("Starting Updates from:", m_positionSource->sourceName().toStdString()));
+    m_positionSource->startUpdates();
+    QGeoPositionInfo info = m_positionSource->lastKnownPosition();
+    m_positionSource->requestUpdate();
+  }
+}
+
+void QtLocationService::Stop()
+{
+  if (!m_positionSource)
+  {
+    LOG(LWARNING, ("No supported source is available for stopping the positioning with:", m_positionSource->sourceName().toStdString()));
+  }
+  else
+  {
+    LOG(LDEBUG, ("Stopping Updates from:", m_positionSource->sourceName().toStdString()));
+    m_positionSource->stopUpdates();
+  }
+}
+
+location::LocationService * CreateQtLocationService(location::LocationObserver & observer, const std::string & sourceName)
+{
+  return new QtLocationService(observer, sourceName);
```

Comment:
nit: It would be great to refactor our location service to use unique_ptr and make_unique instead of raw pointers.

Reply:
I agree. And I also think that such a refactor would impact all the platforms, and should go into a separate PR.
Is that OK?